Update Error Codes.md (#1325)
wix/wix-code-docs: wix-data/wix-data/guides/Error Codes.md

|WDE0044| Invalid {function name} parameter value. {function name} parameter must be a String.| Check that the parameter value indicated in the error message is a string.|
|WDE0052| The current site is in template mode. Please save it to modify data.| To modify collection data on a site created from a template, make sure to [save the site](https://support.wix.com/en/article/wix-editor-save-preview-and-publish-your-site#saving-your-site) first.
|WDE0053| Internal wixData error: {error message}| An unexpected error occurred. If the error persists, please [contact support](https://www.wix.com/contact).

> |WDE0054| Internal server error. wix-data fails to parse the error it gets provided. | This error occurs when you make an invalid request. Check that the data requests in your code are valid.

Reviewer: A general comment for everything added here. The middle column is supposed to contain the actual text of the error code. Just making sure that's the case for all the codes you added.

In terms of this particular request, I think we should provide some context if possible. Which functions or types of requests might cause this?

|WDE0073| Item {requested item's ID} does not exist in collection {your collection name}.| Check that an item with the ID you're requesting exists in the collection you're querying.
|WDE0074| WD_ITEM_ALREADY_EXISTS| An item with the requested ID value already exists. This error occurs when you try to insert an item into a collection with an `_id` value that already exists in that collection. If you are trying to either insert a new item or update it if it already exists, use [`save()`](/wix-data/save).
|WDE0075| WD_BAD_REQUEST| This error occurs when you make an invalid request using the [`aggregate()`](/wix-data/aggregate) function. See the [API reference](/wix-data/wixdataaggregate) for details on building the WixDataAggregate objects used with this function.|
|WDE0076| Validation error. {error message from Wix app collection or external database}|A request to a [Wix app collection](https://support.wix.com/en/article/velo-working-with-wix-app-collections-and-code) or an [external database](https://support.wix.com/en/article/velo-integrating-external-databases-with-your-wix-site) that you're working with returned with an error. Check that the data requests in your code are valid.

> |WDE0077| Provided query limit is greater than supported by the collection. | Provide a lower value.

Reviewer: ```suggestion
|WDE0077| Provided query limit is greater than supported by the collection. | Provide a lower value for the query limit.
```

|WDE0078| Hook error.| The code for one of your [data hook](https://support.wix.com/en/article/velo-about-data-hooks) functions threw an error.<br><br>For example, consider an `afterInsert` data hook function in a `backend/data.js` file that includes this line of code: `throw new Error("sample error message")`<br><br>Once you define this hook, calling the [`insert()`](/wix-data/insert) function results in the `WDE0078` error. 
|WDE0079| {your collection name} is a single item collection and item must have id {expected ID}, but got {received ID} instead.| The item in a single item collection has a fixed `_id` value. Check that your aren't trying to change this value in your code. You can find the item's `_id` value using the [Content Manager](https://support.wix.com/en/article/about-the-content-manager-7160473#collections) or by using [`find()`](/wix-data/wixdataquery/find) without any filters.

> |WDE0080| User request validation error. Incoming API request payload is invalid. | `collectionName` for a given data operation might not be provided. Check the payload data.

Reviewer: ```suggestion
|WDE0080| User request validation error. Incoming API request payload is invalid. | You may not have provided the `collectionName` value for a data operation. Check the payload data.
```

|WDE0092| Dataset is too large to sort.| The data you tried to sort didn't fit into memory. This usually occurs when large items are being sorted or there are too many items being sorted to fit into memory at once. [Add an index](https://support.wix.com/en/article/content-manager-adding-an-index-for-your-collection) to your collection to avoid using up memory.
|WDE0093| Invalid filter for _publishStatus field, only .eq and .ne filters are allowed with 'DRAFT' and 'PUBLISHED' possible values.| When you enable the [Draft and Publish Items](https://support.wix.com/en/article/content-manager-using-the-draft-publish-items-feature) feature for a collection, a `_publishStatus` field is added to every item in that collection. If you want to filter a data request based on this field, you can only use the [`eq()`](/wix-data/wixdatafilter/eq) and [`ne()`](/wix-data/wixdatafilter/ne) functions. The only valid values for `_publishedStatus` are `'DRAFT'` and `'PUBLISHED'`. Check that your code isn't using other filter functions or values with `_publishedStatus`.

> |WDE0110| WD_BAD_REQUEST. | WixCodeNotEnabled - wix-data fails to resolve wix-code context. The error is not expected under normal circumstances, which usually means something is wrong with the system.

Reviewer: ```suggestion
|WDE0110| WD_BAD_REQUEST. | This code is likely the result of a system error.
```

> |WDE0111| User request validation error. | `GridAppId` parameter must be provided in the SANDBOX segment.

Reviewer: Is this relevant for Velo users?

|WDE0115| WD_UNKNOWN_ERROR| Your external database requested an unsupported data operation.
|WDE0116| WD_UNKNOWN_ERROR| Your external database responded to Wix with an error.

> |WDE0120| WD_BAD_REQUEST| Unsupported Driver Operation - `$collectionName` does not support `$operation` operation.

Reviewer: Two questions:

1. Is this relevant for Velo users?
2. The "troubleshooting" reads like an error message. Are you sure it's not the error message itself?

|WDE0121| Sorting by multiple array fields is not supported| If there are multiple fields in your collection that contain arrays, you can't sort by both of them at once.<br><br>For example, consider a collection item that includes a field called `A` whose value is `[1,2,3]` and a field called `B` whose value is `[3,2,1]`. In this case, attempting to sort by both `A` and `B` at once results in an error. If your collection contains multiple fields that are arrays, check that you aren't trying to sort using more than one at once.
|WDE0122| Processing time quota exceeded.| Wix limits the number of read and write requests you can make from your site per minute. The maximum number for read requests is 3000 and the maximum for write requests is 1500. We can increase these quotas in certain cases. [Learn More](https://support.wix.com/en/article/velo-about-data-quotas#overall-processing-time) about this quota and what you can do to resolve this issue.

> |WDE0123| WD_ITEM_ALREADY_EXISTS| Unique index conflict error. Possible duplicate value.

Reviewer: For the troubleshooting, I think we need to clarify when this error is triggered (i.e., which functions cause it etc.) Also, what is being duplicated? Is it the `_id` field?

> |WDE0128| Adapter `ItemId` is missing. | Item must have an `_id` field. 

Reviewer: ```suggestion
|WDE0128| Adapter `ItemId` is missing. | Check that each item has an `_id` field. 
```

> |WDE0133| Document is too large to index.| 

Reviewer: We need some kind of troubleshooting text, even if it it's just filler.

> |WDE0134| Dollar($) prefix field names can not be written to the database. | Remove dollar($) prefix from the request.

Reviewer: What context is this happening in? Is it for external DBs? I think it might be useful to mention that in the troubleshooting description.

```suggestion
|WDE0134| Dollar($) prefix field names can not be written to the database. | Make sure that none of your field names in your request have a `$` prefix.
```
